lib/gem2rpm.rb

```diff
@@ -90,7 +90,7 @@ def self.convert(fname, template, out = $stdout, nongem = true, local = false,
       end
     end
 
-    erb = ERB.new(template.read, 0, '-')
+    erb = ERB.new(template.read, trim_mode: '-')
```

Comment:
This raises the minimum version to Ruby 2.7. Doesn't have to be a problem, but good to remember. I'm not a maintainer so I don't know if they want to keep Ruby < 2.7 support.

Reply:
This is good point. Ruby 2.0.0 should be supported as long as RHEL 7 is supported. This is the support matrix:

https://github.com/fedora-ruby/gem2rpm/blob/master/.travis.yml

However, not sure in what state Travis is ATM :/

Reply:
Travis can be considered dead for open source projects. GitHub Actions are quite trivial to use though: https://github.com/fedora-ruby/gem2rpm/pull/112

Reply:
Should I add a switch depending on the ruby version? RHEL 7 is going to exist for a while, but newer ruby versions will probably deprecate the current behavior eventually

Reply:
That's what I did. What I used in other projects:
```ruby
erb = if RUBY_VERSION >= '2.7'
  ERB.new(template.read, trim_mode: '-')
else
  ERB.new(template.read, 0, '-')
end
```

I'm not sure in which versions it's valid, but at least >= 2.7 supported the newer style.

Reply:
Done, thanks!

Reply:
Good idea to branch the code :+1: Nevertheless, is there a way to use Ruby introspection instead? I don't like the version checks, but if there is no other way around, then this is acceptable indeed. Sorry for being demanding :)

Reply:
I further looked into this and I'm not sure why I thought it only worked in Ruby 2.7. Ruby 2.0 introduced keyword arguments and I just checked that `ERB.new('', trim_mode: '-')` works in Ruby 2.0 as well. Sorry for the confusion.

Reply:
I have reverted the change and rebased this branch on master. @voxik could you allow the workflow so that the CI will run?

Reply:
@voxik you can change the GH settings to allow actions to run. The default setting is to allow approval for first time committers, but you can relax this to "Require approval for first-time contributors who are new to GitHub". This is under settings -> actions -> general.

Reply:
Not really sure what is the result of those steps, but I did them :wink:

Reply:
And clicked "approve and run", so the actions should be executed now I believe

Reply:
The tests are red on Ruby 2.5 and older. Please have a look. Perhaps I was right, but incorrect on the specific versions?

Reply:
I checked and https://www.ruby-lang.org/en/news/2018/12/25/ruby-2-6-0-released/ states it then turned into a keyword argument. So a `RUBY_VERSION >= 2.6` check is correct.

Reply:
So how about checking this way:

~~~
ERB.new("").method(:initialize).parameters & [[:key, :trim_mode]]
~~~

Reply:
This check seems to be universal. However, some comment explaining that this could be removed as soon as Ruby 2.6+ is supported would probably not hurt.

Reply:
Just FTR:

https://github.com/fedora-ruby/gem2rpm/pull/117